packages/slice-machine/components/Button/index.tsx

```diff
@@ -2,39 +2,72 @@ import React from "react";
 
 import { Button as ThemeUIButton, Spinner } from "theme-ui";
 import { ThemeUIStyleObject } from "@theme-ui/css";
+import { IconType } from "react-icons";
 
-export type SliceMachineButtonProps = {
-  form?: string;
+export type ButtonProps = {
+  label: string;
+  Icon?: IconType;
   type?: "submit" | "reset" | "button";
+  form?: string;
   isLoading?: boolean;
   disabled?: boolean;
   onClick?: React.MouseEventHandler<HTMLButtonElement>;
   sx?: ThemeUIStyleObject;
+  "data-cy"?: string;
+  variant?: string;
 };
 
-const Button: React.FunctionComponent<SliceMachineButtonProps> = ({
+// Small helper to allow us to target spinner and icon in the CY
+const cyIdBuilder = (dataCy: string | undefined, id: string) => {
+  if (dataCy) return `${dataCy}-${id}`;
+  return "";
+};
+
+// If you don't use an icon, don't forget to pass a min-width property so the button doesn't change width on loading.
```

Comment:
I see what you mean, good point

Reply:
Ok so I implemented it and then realise there are some issues

- the button is used in the modal card on which we know nothing of the height of the button as the label is passed down through a property. Those are not buttons that were required to changed, but they used the previous component so i tried to keep the same behavior.
<img width="393" alt="image" src="https://user-images.githubusercontent.com/41187899/194839025-1b284f1f-92f6-4304-8427-ee5c0fb024a9.png">

- For now there is only one place where we need to use the width property it's in `packages/slice-machine/lib/builders/SliceBuilder/SetupDrawer/steps/common.tsx`. It actually requires a max-width and not min-width because the button renders on the whole width of the parent. So it's not even the use case where we could need min-width to make sure the button doesn't glitch.

Reply:
I can set a minimum for the first case but it would still glitch. (as there are different label -> button width that i cannot predict).
For the second case, I tried having types like you suggested, it add quite a lot of complexity and I didn't manage to make the type works unless having a property with a special type.

Reply:
Makes sense 👍 . Probably not worth digging into much more anyway since the bug should be pretty obvious on first use if the developer misses it 😅